File: ios/Gather/Theme/Theme.swift
import SwiftUI

/// Gather — dark-only, Netflix-inspired theme.
/// Deep black canvas, contrasted poster cards, vivid red accent, rare gold highlights.
nonisolated enum Palette {
    // Canvas
    static let bg = Color(hex: 0x141416)
    static let bgDeep = Color(hex: 0x0B0B0C)
    static let bgElevated = Color(hex: 0x1B1B1E)

    // Surfaces
    static let card = Color(hex: 0x1C1C1F)
    static let cardElevated = Color(hex: 0x242428)
    static let surface = Color(hex: 0x2A2A2E)

    // Lines
    static let border = Color(hex: 0x303034)
    static let borderStrong = Color(hex: 0x3D3D42)

    // Text
    static let text = Color(hex: 0xF7F7F7)
    static let textMuted = Color(hex: 0x9B9BA2)
    static let textFaint = Color(hex: 0x67676E)

    // Brand
    static let primary = Color(hex: 0xEF233C)
    static let primaryDark = Color(hex: 0xC2182C)
    static let primarySoft = Color(hex: 0xEF233C).opacity(0.14)
    static let primaryFg = Color.white

    // Highlights
    static let accent = Color(hex: 0xCDBE57)
    static let accentSoft = Color(hex: 0xCDBE57).opacity(0.16)
    static let goldFg = Color(hex: 0x1A1607)

    // Status
    static let success = Color(hex: 0x3DD27E)
    static let successSoft = Color(hex: 0x3DD27E).opacity(0.14)
    static let warning = Color(hex: 0xE8A13A)
    static let destructive = Color(hex: 0xE63946)
    static let destructiveSoft = Color(hex: 0xE63946).opacity(0.16)
}

nonisolated enum Radius {
    static let sm: CGFloat = 8
    static let md: CGFloat = 10
    static let lg: CGFloat = 12
    static let xl: CGFloat = 16
    static let xxl: CGFloat = 22
    static let pill: CGFloat = 999
}

nonisolated enum Spacing {
    static let xs: CGFloat = 4
    static let sm: CGFloat = 8
    static let md: CGFloat = 12
    static let lg: CGFloat = 16
    static let xl: CGFloat = 20
    static let xxl: CGFloat = 28
    static let xxxl: CGFloat = 40
}

// MARK: - Typography

nonisolated enum TypeStyle {
    case display, title, h2, h3, body, bodyMuted, label, caption, overline

    var size: CGFloat {
        switch self {
        case .display: return 33
        case .title: return 26
        case .h2: return 20
        case .h3: return 17
        case .body, .bodyMuted: return 15
        case .label: return 13
        case .caption: return 12
        case .overline: return 11
        }
    }

    var weight: Font.Weight {
        switch self {
        case .display, .title: return .heavy
        case .h2, .h3, .overline: return .bold
        case .label, .caption: return .semibold
        case .body, .bodyMuted: return .medium
        }
    }

    var tracking: CGFloat {
        switch self {
        case .display: return -0.8
        case .title: return -0.6
        case .h2: return -0.3
        case .h3: return -0.2
        case .overline: return 1.4
        case .caption: return 0.2
        default: return 0
        }
    }

    var color: Color {
        switch self {
        case .bodyMuted, .label: return Palette.textMuted
        case .caption, .overline: return Palette.textFaint
        default: return Palette.text
        }
    }

    var uppercased: Bool { self == .overline }
}

private struct TypeModifier: ViewModifier {
    let style: TypeStyle
    func body(content: Content) -> some View {
        content
            .font(.system(size: style.size, weight: style.weight))
            .tracking(style.tracking)
            .textCase(style.uppercased ? .uppercase : nil)
            .foregroundStyle(style.color)
    }
}

extension View {
    func gType(_ style: TypeStyle) -> some View { modifier(TypeModifier(style: style)) }
}

// MARK: - Shadows

extension View {
    func posterShadow() -> some View { shadow(color: .black.opacity(0.55), radius: 22, x: 0, y: 16) }
    func cardShadow() -> some View { shadow(color: .black.opacity(0.4), radius: 13, x: 0, y: 8) }
    func glowShadow(_ active: Bool = true) -> some View {
        shadow(color: active ? Palette.primary.opacity(0.45) : .clear, radius: active ? 18 : 0)
    }
}

// MARK: - Color hex

extension Color {
    init(hex: UInt, alpha: Double = 1) {
        self.init(
            .sRGB,
            red: Double((hex >> 16) & 0xFF) / 255,
            green: Double((hex >> 8) & 0xFF) / 255,
            blue: Double(hex & 0xFF) / 255,
            opacity: alpha
        )
    }
}

File: expo/constants/theme.ts
import { Platform, TextStyle, ViewStyle } from "react-native";

/**
 * Gather — dark-only, Netflix-inspired theme.
 * Deep black canvas, contrasted poster cards, vivid red accent, rare gold highlights.
 */
export const colors = {
  // Canvas
  bg: "#141416",
  bgDeep: "#0B0B0C",
  bgElevated: "#1B1B1E",

  // Surfaces
  card: "#1C1C1F",
  cardElevated: "#242428",
  surface: "#2A2A2E",

  // Lines
  border: "#303034",
  borderStrong: "#3D3D42",

  // Text
  text: "#F7F7F7",
  textMuted: "#9B9BA2",
  textFaint: "#67676E",

  // Brand
  primary: "#EF233C",
  primaryDark: "#C2182C",
  primarySoft: "rgba(239,35,60,0.14)",
  primaryFg: "#FFFFFF",

  // Highlights
  accent: "#CDBE57",
  accentSoft: "rgba(205,190,87,0.16)",

  // Status
  success: "#3DD27E",
  successSoft: "rgba(61,210,126,0.14)",
  warning: "#E8A13A",
  destructive: "#E63946",
  destructiveSoft: "rgba(230,57,70,0.16)",

  // Misc
  overlay: "rgba(0,0,0,0.6)",
  scrim: "rgba(10,10,12,0.92)",
  white: "#FFFFFF",
  black: "#000000",
} as const;

export const radius = {
  sm: 8,
  md: 10,
  lg: 12,
  xl: 16,
  xxl: 22,
  pill: 999,
} as const;

export const spacing = {
  xs: 4,
  sm: 8,
  md: 12,
  lg: 16,
  xl: 20,
  xxl: 28,
  xxxl: 40,
} as const;

const fontFamily = Platform.select({ ios: "System", android: "sans-serif", default: "System" });

export const fonts = {
  black: Platform.select({ ios: "System", android: "sans-serif", default: "System" }),
  family: fontFamily,
} as const;

export const type = {
  display: {
    fontSize: 34,
    lineHeight: 38,
    fontWeight: "800" as const,
    letterSpacing: -0.8,
    color: colors.text,
  } satisfies TextStyle,
  title: {
    fontSize: 26,
    lineHeight: 30,
    fontWeight: "800" as const,
    letterSpacing: -0.6,
    color: colors.text,
  } satisfies TextStyle,
  h2: {
    fontSize: 20,
    lineHeight: 25,
    fontWeight: "700" as const,
    letterSpacing: -0.3,
    color: colors.text,
  } satisfies TextStyle,
  h3: {
    fontSize: 17,
    lineHeight: 22,
    fontWeight: "700" as const,
    letterSpacing: -0.2,
    color: colors.text,
  } satisfies TextStyle,
  body: {
    fontSize: 15,
    lineHeight: 21,
    fontWeight: "500" as const,
    color: colors.text,
  } satisfies TextStyle,
  bodyMuted: {
    fontSize: 15,
    lineHeight: 21,
    fontWeight: "500" as const,
    color: colors.textMuted,
  } satisfies TextStyle,
  label: {
    fontSize: 13,
    lineHeight: 17,
    fontWeight: "600" as const,
    color: colors.textMuted,
  } satisfies TextStyle,
  caption: {
    fontSize: 12,
    lineHeight: 15,
    fontWeight: "600" as const,
    letterSpacing: 0.2,
    color: colors.textFaint,
  } satisfies TextStyle,
  overline: {
    fontSize: 11,
    lineHeight: 14,
    fontWeight: "700" as const,
    letterSpacing: 1.4,
    textTransform: "uppercase" as const,
    color: colors.textFaint,
  } satisfies TextStyle,
} as const;

export const shadows = {
  poster: {
    shadowColor: "#000000",
    shadowOpacity: 0.55,
    shadowRadius: 26,
    shadowOffset: { width: 0, height: 16 },
    elevation: 14,
  } satisfies ViewStyle,
  card: {
    shadowColor: "#000000",
    shadowOpacity: 0.4,
    shadowRadius: 14,
    shadowOffset: { width: 0, height: 8 },
    elevation: 8,
  } satisfies ViewStyle,
  glow: {
    shadowColor: colors.primary,
    shadowOpacity: 0.5,
    shadowRadius: 18,
    shadowOffset: { width: 0, height: 0 },
    elevation: 10,
  } satisfies ViewStyle,
} as const;

export const theme = { colors, radius, spacing, type, shadows, fonts } as const;
export type Theme = typeof theme;
